Sources/SwiftDocC/Infrastructure/Symbol Graph/ExtendedTypesFormatTransformation.swift

```diff
@@ -0,0 +1,530 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import SymbolKit
+
+/// A namespace comprising functionality for converting between the standard Symbol Graph File
+/// format with extension block symbols and the extended types format extension used by SwiftDocC.
+enum ExtendedTypesFormatTransformation { }
+
+extension ExtendedTypesFormatTransformation {
+    /// Convert from the extension block symbol format to the extended type symbol format.
+    ///
+    /// First, the function checks if there are any symbols of kind `.extension` in the graph.
+    /// If not, function returns `false` without altering the graph in any way.
+    ///
+    /// If it finds such symbols, it applies the actual transformation. Refer to the sections below to find
+    /// out how the two formats differ.
+    ///
+    /// In addition, the transformation prepends the given `moduleName` to the `pathComponents` of all
+    /// symbols in the graph.
+    ///
+    /// ### The Extension Block Symbol Format
```

Comment:
Great documentation.

Reply:
Minor: a few words here and there in this file have typos in them:

- accross
- anchestor
- closue
- documentaiton
- extemdedClass
- extenison
- extensoin
- follwing
- fragement

Reply:
🙈 Thanks! Fixed in 89fd7c23bff21602fab18abfebeee60ceb0efc56.